src/actions/index.js

```diff
@@ -81,6 +82,30 @@ const initialize = () => {
   };
 };
 
+// This code defends against the case where /expression returns a cellname
+// never seen before (ie, not returned by /cells).   This should not happen
+// (see https://github.com/chanzuckerberg/cellxgene-rest-api/issues/34) but
+// occasionally does.
+//
+function cleanupExpressionResponse(data) {
+  const s = store.getState();
+  const metadata = s.controls.currentCellSelectionMap;
+  let errorOccured = false;
+  const newcells = _.filter(data.data.cells, cell => {
+    const found = metadata[cell.cellname];
+    errorOccured = errorOccured || !found;
```

Comment:
This is acceptable but reads slightly obfuscated to me.

Reply:
happy to fix.   what is obfuscated?  Is it the inversion of the boolean?   (eg, switch to using `notFound`?)

Reply:
That change would print a console error for every missing cell (seems excessive :-).

How about this:

```js
  let errorLogged = false;
  const newcells = _.filter(data.data.cells, cell => {
    if (!errorLogged && !metadata[cell.cellname]) {
      errorLogged = true;
      console.error(
        "Warning: /expression REST API returned unexpected cell names -- discarding surprises."
      );
      return !metadata[cell.cellname];
    }
  });
```

Reply:
OK, just sent you a PR containing fixes.